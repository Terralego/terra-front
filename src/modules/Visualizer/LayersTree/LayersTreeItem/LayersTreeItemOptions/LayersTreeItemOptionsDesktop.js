import React from 'react';
import classnames from 'classnames';
import { Button } from '@blueprintjs/core';

import translateMock from '../../../../../utils/translate';

import LayerFetchValues from '../LayerFetchValues';
import FiltersPanel from '../FiltersPanel';
import Tooltip from '../../../../../components/Tooltip';

const LayersTreeItemOptionsDesktop = ({
  hasSomeOptionActive,
  isOptionsOpen,
  handleOptionPanel,
  layer,
  toggleFilters,
  isFilterVisible,
  getFilterPanelRef,
  form,
  toggleTable,
  isTableActive,
  displayTableButton,
  toggleWidgets,
  widgets,
  isWidgetActive,
  translate = translateMock({
    'terralego.visualizer.layerstree.itemOptions.widget.title': 'widget',
    'terralego.visualizer.layerstree.itemOptions.widget.action-open': 'open widget',
    'terralego.visualizer.layerstree.itemOptions.widget.action-close': 'close widget',
    'terralego.visualizer.layerstree.itemOptions.widget.title_synthesis': 'widget synthesis',
    'terralego.visualizer.layerstree.itemOptions.widget.action-open_synthesis': 'open synthesis',
    'terralego.visualizer.layerstree.itemOptions.widget.action-close_synthesis': 'close synthesis',
    'terralego.visualizer.layerstree.itemOptions.table.label': 'table',
    'terralego.visualizer.layerstree.itemOptions.table.alt': 'open table',
    'terralego.visualizer.layerstree.itemOptions.table.alt_close': 'close table',
    'terralego.visualizer.layerstree.itemOptions.table.tooltip': 'open table',
    'terralego.visualizer.layerstree.itemOptions.table.tooltip_close': 'close table',
    'terralego.visualizer.layerstree.itemOptions.filter.label': 'filters',
    'terralego.visualizer.layerstree.itemOptions.filter.alt': 'open filters',
    'terralego.visualizer.layerstree.itemOptions.filter.alt_close': 'close filters',
    'terralego.visualizer.layerstree.itemOptions.filter.tooltip': 'open filters',
    'terralego.visualizer.layerstree.itemOptions.filter.tooltip_close': 'close filters',
    'terralego.visualizer.layerstree.itemOptions.options.label': 'options',
    'terralego.visualizer.layerstree.itemOptions.options.alt': 'open options',
    'terralego.visualizer.layerstree.itemOptions.options.alt_close': 'close options',
    'terralego.visualizer.layerstree.itemOptions.options.tooltip': 'open options',
    'terralego.visualizer.layerstree.itemOptions.options.tooltip_close': 'close options',
  }),
}) => (
  <div
    className={classnames(
      'layerstree-node-content__options',
      'layerstree-node-content__options--desktop',
      { 'layerstree-node-content__options--active': hasSomeOptionActive },
    )}
  >
    {(widgets && !!widgets.length) && (
      // i18next-extract-mark-context-start ["", "synthesis"]
      widgets.map(widget => {
        const { component: context } = widget;
        const isActive = isWidgetActive(widget);

        const actionText = isActive
          ? translate('terralego.visualizer.layerstree.itemOptions.widget.action-close', { context })
          : translate('terralego.visualizer.layerstree.itemOptions.widget.action-open', { context });

        return (
          <Tooltip
            key={context}
            className="layerstree-node-content__options__tooltip widgets"
            content={actionText}
          >
            <Button
              className={classnames({
                'layerstree-node-content__options__button': true,
                'layerstree-node-content__options__button--active': isActive,
              })}
              onClick={toggleWidgets(widget)}
              minimal
              icon="selection"
              title={translate('terralego.visualizer.layerstree.itemOptions.widget.title', {
                context,
              })}
              alt={actionText}
            />
          </Tooltip>
        );
      })
      // i18next-extract-mark-context-stop
    )}
    {displayTableButton && (
      <Tooltip
        className="layerstree-node-content__options__tooltip table"
        content={translate('terralego.visualizer.layerstree.itemOptions.table.tooltip', {
          context: isTableActive ? 'close' : 'open',
        })}
      >
        <Button
          className={
        classnames(
          'layerstree-node-content__options__button',
          { 'layerstree-node-content__options__button--active': isTableActive },
        )
      }
          onClick={toggleTable}
          minimal
          icon="th"
          title={translate('terralego.visualizer.layerstree.itemOptions.table.title')}
          alt={translate('terralego.visualizer.layerstree.itemOptions.table.alt', {
            context: isTableActive ? 'close' : 'open',
          })}
        />
      </Tooltip>
    )}
    {form && (
    <FiltersPanel
      visible={isFilterVisible}
      onMount={getFilterPanelRef}
      layer={layer}
    >
      {isFilterVisible && (
      <LayerFetchValues layer={layer} isFilterVisible={isFilterVisible} />
      )}
      <Tooltip
        content={translate('terralego.visualizer.layerstree.itemOptions.filter.tooltip', {
          context: isFilterVisible ? 'close' : 'open',
        })}
        className="visualizer.layerstree-node-content__options__tooltip filters"
      >
        <Button
          className={
        classnames(
          'layerstree-node-content__options__button',
          { 'layerstree-node-content__options__button--active': isFilterVisible },
        )}
          onClick={toggleFilters}
          minimal
          icon="filter"
          title={translate('terralego.visualizer.layerstree.itemOptions.filter.label')}
          alt={translate('terralego.visualizer.layerstree.itemOptions.filter.alt', {
            context: isFilterVisible ? 'close' : 'open',
          })}
        />
      </Tooltip>
    </FiltersPanel>
    )}
    <Tooltip
      content={translate('terralego.visualizer.layerstree.itemOptions.options.tooltip', {
        context: isOptionsOpen ? 'close' : 'open',
      })}

      className="layerNode__tooltip options"
    >
      <Button
        className={
          classnames(
            'layerstree-node-content__options__button',
            'layerstree-node-content__options__button--more',
            { 'layerstree-node-content__options__button--active': isOptionsOpen },
          )}
        icon="more"
        minimal
        onClick={handleOptionPanel}
        title={translate('terralego.visualizer.layerstree.itemOptions.options.label')}
        alt={translate('terralego.visualizer.layerstree.itemOptions.options.alt', {
          context: isOptionsOpen ? 'close' : 'open',
        })}
      />
    </Tooltip>
  </div>
);

export default LayersTreeItemOptionsDesktop;
