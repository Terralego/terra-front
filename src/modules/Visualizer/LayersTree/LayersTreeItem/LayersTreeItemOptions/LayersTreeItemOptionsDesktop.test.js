import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import LayersTreeItemOptionsDesktop from './LayersTreeItemOptionsDesktop';

jest.mock('../LayerFetchValues', () => function LayerFetchValues () {
  return <p>LayerFetchValues</p>;
});

jest.mock('../FiltersPanel', () => function FiltersPanel () {
  return <p>FiltersPanel</p>;
});

it('should render default options', () => {
  const tree = renderer.create((
    <LayersTreeItemOptionsDesktop />));
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render with properties', () => {
  const tree = renderer.create((
    <LayersTreeItemOptionsDesktop
      displayTableButton
      form={[]}
      widgets={[{ component: 'foo' }]}
      isWidgetActive={jest.fn(() => true)}
      toggleWidgets={jest.fn(() => true)}
    />
  ));
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render no tooltip if empty widget empty', () => {
  const tree = renderer.create((
    <LayersTreeItemOptionsDesktop
      widgets={[]}
    />
  ));
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should open widget', () => {
  const wrapper = shallow((
    <LayersTreeItemOptionsDesktop
      displayTableButton
      form={[]}
      widgets={[{ component: 'foo' }]}
      isWidgetActive={jest.fn(() => false)}
      toggleWidgets={jest.fn(() => true)}
    />
  ));
  expect(wrapper.find('.widgets').props().content).toBe('open widget');
});

it('should close table', () => {
  const wrapper = shallow((
    <LayersTreeItemOptionsDesktop
      displayTableButton
      isTableActive
    />
  ));
  expect(wrapper.find('.table').props().content).toBe('close table');
  expect(wrapper.find('.table').dive().find('.layerstree-node-content__options__button').props().alt).toBe('close table');
});

it('should close form', () => {
  const wrapper = shallow((
    <LayersTreeItemOptionsDesktop
      form={[]}
      isFilterVisible
    />
  ));
  expect(wrapper.find('.filters').props().content).toBe('close filters');
});

it('should close options', () => {
  const wrapper = shallow((
    <LayersTreeItemOptionsDesktop
      isOptionsOpen
    />
  ));
  expect(wrapper.find('.options').props().content).toBe('close options');
  expect(wrapper.find('.options').dive().find('.layerstree-node-content__options__button').props().alt).toBe('close options');
});
