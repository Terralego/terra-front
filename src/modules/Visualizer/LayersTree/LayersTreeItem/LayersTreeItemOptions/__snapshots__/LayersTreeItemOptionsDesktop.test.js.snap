// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default options 1`] = `
<div
  className="layerstree-node-content__options layerstree-node-content__options--desktop"
>
  <span
    className="bp3-popover-wrapper layerNode__tooltip options"
  >
    <span
      className="bp3-popover-target"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        alt="open options"
        className="bp3-button bp3-minimal layerstree-node-content__options__button layerstree-node-content__options__button--more"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="options"
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-more"
          icon="more"
        >
          <svg
            data-icon="more"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              more
            </desc>
            <path
              d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
</div>
`;

exports[`should render no tooltip if empty widget empty 1`] = `
<div
  className="layerstree-node-content__options layerstree-node-content__options--desktop"
>
  <span
    className="bp3-popover-wrapper layerNode__tooltip options"
  >
    <span
      className="bp3-popover-target"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        alt="open options"
        className="bp3-button bp3-minimal layerstree-node-content__options__button layerstree-node-content__options__button--more"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="options"
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-more"
          icon="more"
        >
          <svg
            data-icon="more"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              more
            </desc>
            <path
              d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
</div>
`;

exports[`should render with properties 1`] = `
<div
  className="layerstree-node-content__options layerstree-node-content__options--desktop"
>
  <span
    className="bp3-popover-wrapper layerstree-node-content__options__tooltip widgets"
  >
    <span
      className="bp3-popover-target"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        alt="close widget"
        className="bp3-button bp3-minimal layerstree-node-content__options__button layerstree-node-content__options__button--active"
        onClick={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="widget"
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-selection"
          icon="selection"
        >
          <svg
            data-icon="selection"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              selection
            </desc>
            <path
              d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-9C6.34 5 5 6.34 5 8s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
  <span
    className="bp3-popover-wrapper layerstree-node-content__options__tooltip table"
  >
    <span
      className="bp3-popover-target"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        alt="open table"
        className="bp3-button bp3-minimal layerstree-node-content__options__button"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="terralego.visualizer.layerstree.itemOptions.table.title"
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-th"
          icon="th"
        >
          <svg
            data-icon="th"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              th
            </desc>
            <path
              d="M15 1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V2c0-.5-.4-1-1-1zM6 13H2v-2h4v2zm0-3H2V8h4v2zm0-3H2V5h4v2zm8 6H7v-2h7v2zm0-3H7V8h7v2zm0-3H7V5h7v2z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
  <p>
    FiltersPanel
  </p>
  <span
    className="bp3-popover-wrapper layerNode__tooltip options"
  >
    <span
      className="bp3-popover-target"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        alt="open options"
        className="bp3-button bp3-minimal layerstree-node-content__options__button layerstree-node-content__options__button--more"
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        tabIndex={0}
        title="options"
        type="button"
      >
        <span
          className="bp3-icon bp3-icon-more"
          icon="more"
        >
          <svg
            data-icon="more"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <desc>
              more
            </desc>
            <path
              d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </button>
    </span>
  </span>
</div>
`;
